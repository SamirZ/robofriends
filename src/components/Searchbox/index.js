import React from 'react';

const SearchBox = ({ searchChange }) => (
  <div className='pa2'>
    <input
      onChange={searchChange}
      className='pa3 ba b--green bg-lightest-blue'
      type='search'
      placeholder='Search robots'
    />
  </div>
);

export default SearchBox;
