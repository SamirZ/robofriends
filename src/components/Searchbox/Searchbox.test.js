import React from 'react';
import { shallow } from 'enzyme';
import Searchbox from './index';

it('expect to render Searchbox component', () => {
  expect(shallow(<Searchbox />)).toMatchSnapshot();
});
