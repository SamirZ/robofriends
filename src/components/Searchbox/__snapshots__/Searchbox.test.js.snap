// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Searchbox component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Searchbox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="pa2 b lightest-blue"
          htmlFor="search"
        >
          Search robots
        </label>,
        <input
          className="pa3 ba b--green bg-lightest-blue mw5"
          id="search"
          placeholder="Search robots"
          type="search"
        />,
      ],
      "className": "pa2 flex flex-column justify-center items-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search robots",
          "className": "pa2 b lightest-blue",
          "htmlFor": "search",
        },
        "ref": null,
        "rendered": "Search robots",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "pa3 ba b--green bg-lightest-blue mw5",
          "id": "search",
          "onChange": undefined,
          "placeholder": "Search robots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="pa2 b lightest-blue"
            htmlFor="search"
          >
            Search robots
          </label>,
          <input
            className="pa3 ba b--green bg-lightest-blue mw5"
            id="search"
            placeholder="Search robots"
            type="search"
          />,
        ],
        "className": "pa2 flex flex-column justify-center items-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search robots",
            "className": "pa2 b lightest-blue",
            "htmlFor": "search",
          },
          "ref": null,
          "rendered": "Search robots",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "pa3 ba b--green bg-lightest-blue mw5",
            "id": "search",
            "onChange": undefined,
            "placeholder": "Search robots",
            "type": "search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
